### PR TITLE
feat: Simplify configs for enabling/disabling operators

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

Simplify configs so that there is one "enabled" config per oeprator, defaulting to true. We can disable operators by setting to false.

### CODE REVIEW

**Review by viirya — APPROVED** [review]

File: spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala [hunk 1]
```diff
@@ -1240,7 +1235,7 @@ class CometExecSuite extends CometTestBase {
         .saveAsTable("bucketed_table2")
 
       withSQLConf(
-        "spark.comet.exec.sortMergeJoin.disabled" -> "true",
+        CometConf.COMET_EXEC_SORT_MERGE_JOIN_ENABLED.key -> "false",
```
Comment (comphead) [inline spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:1238]: true -> false change, is it related?

Comment (viirya) [reply spark/src/test/scala/org/apache/comet/exec/CometExecSuite.scala:1238]: This is to disable sort merge join. Previously it is a `disabled` config. Now it is `enabled` config.

**Review by parthchandra — APPROVED** [review]
lgtm

**Comment by viirya** [comment]
Thanks @andygrove @parthchandra @comphead

**Pull request merged**